### PR TITLE
MDS-3760: Sort the PP

### PR DESCRIPTION
**Pull request by exzizt**
# Main
- Allows users to sort documents in the PP by using a handy "drag-n-drop" feature:
image
  - Sorting is only allowed in the Draft tab and while the draft is in progress
  - The order starts at "1.2" in the table because 1.1 is currently always the Application Form in the permit
 - Accounts for various edge cases, e.g., when a user uploads a document and sets it to "Is In Final Application Package" or when we automatically add the NoW form to the PP we need to add it to the bottom of the PP

### CODE REVIEW

File: services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
-import React from "react";
+import React, { Component } from "react";
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js:1]: I turned this component into a class component to make it easier to use the state (+ it's what the Ant Design demo was using).

File: services/core-web/src/components/noticeOfWork/applications/FinalPermitDocuments.js [hunk 2]
```diff
@@ -111,6 +117,7 @@ export class FinalPermitDocuments extends Component {
           editPreambleFileMetadata={this.props.editPreambleFileMetadata}
           isFinalPackageTable
           isAdminView
+          isSortingAllowed={!this.props.adminView && draftInProgress}
         />
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/FinalPermitDocuments.js:121]: Is sorting is enabled on the manage documents tab, draft tab, but not the admin tab?

File: services/core-api/app/api/now_applications/resources/now_application_export_resource.py [hunk 3]
```diff
@@ -185,20 +185,47 @@
 ]
```
Comment (mmumby) [inline services/core-api/app/api/now_applications/resources/now_application_export_resource.py:185]: I should get the same formatter as you and/or figure out why mine isn't working. (considering I was the last to update this file)

File: services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js [hunk 4]
```diff
@@ -188,7 +306,20 @@ export const NOWDocuments = (props) => {
...
+    const sortColumn = {
+      title: "Order",
+      dataIndex: "index",
+      className: "drag-visible",
+      render: (text) => (
+        <>
+          <DragHandle />
+          {/* NOTE: We are adding 2 here because "1.1" in the issued permits is currently always the application form document. */}
+          &nbsp; 1.{text + 2}
+        </>
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js:318]: order starts at 0?

**Review by mmumby — DISMISSED** [review]
Looks great! 
🔥 

I just have a few questions but nothing jumped out at me.

File: services/core-web/src/components/noticeOfWork/applications/FinalPermitDocuments.js (hunk 2 above)
Comment (exzizt) [reply services/core-web/src/components/noticeOfWork/applications/FinalPermitDocuments.js:121]: Enabled just on the Draft tab. The adminView prop is set to true for the Manage Docs and Admin Tab.

File: services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js (hunk 4 above)
Comment (exzizt) [reply services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js:318]: We use the index (starting from 0) and then need to add 2 to get "1.2".

**Comment by exzizt** [comment]
> Looks great!
> 🔥
> 
> I just have a few questions but nothing jumped out at me.

Thank you for reviewing! 😃

**Review by mmumby — DISMISSED** [review]

**Pull request merged**